flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DefaultDeclarativeSlotPool.java

```diff
@@ -127,7 +155,28 @@ public void increaseResourceRequirementsBy(ResourceCounter increment) {
         }
         totalResourceRequirements = totalResourceRequirements.add(increment);
 
+        if (slotRequestMaxInterval == null) {
+            declareResourceRequirements();
+            return;
+        }
+
+        Preconditions.checkNotNull(componentMainThreadExecutor);
+        if (slotRequestMaxIntervalTimeoutCheckFuture != null) {
+            slotRequestMaxIntervalTimeoutCheckFuture.cancel(true);
+        }
+        slotRequestMaxIntervalTimeoutCheckFuture =
+                componentMainThreadExecutor.schedule(
+                        this::checkSlotRequestMaxIntervalTimeout,
+                        slotRequestMaxInterval.toMilliseconds(),
+                        TimeUnit.MILLISECONDS);
+    }
+
+    private void checkSlotRequestMaxIntervalTimeout() {
+        if (componentMainThreadExecutor == null || slotRequestMaxInterval == null) {
```

Comment:
I think we need to assert something went wrong instead of ignore this illegal statement.

Reply:
nice idea!

How about `When assigning values in the construction method, we do some checks and only allow these two parameters to be both empty or not empty at the same time` ? Or use `Precondition.checkState/Args` to check it ?

Because we only need this `componentMainThreadExecutor ` when allowing requests resources by batch.

Looking forward to better handlings.

Reply:
`Precondition.checkNotNull` on two variables would be good enough from my side.